### PR TITLE
[Analysis] 'when' with subject

### PR DESCRIPTION
**Pull request by serras**

### CODE REVIEW

**Comment by 47degdev** [comment]
Task linked: [CU-pzdybt Handle `when` conditions](https://app.clickup.com/t/pzdybt)

File: plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt [hunk 1]
```diff
@@ -187,6 +186,22 @@ class LiquidDataflowTests {
     )
   }
 
+  @Test
+  fun `when with val`() {
+    """
+      ${imports()}
+      fun bar(x: Int): Int {
+        pre( x >= 0 ) { "x is >= 0" }
+        return (when (val y = x + 1) {
```
Comment (raulraja) [inline plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt:195]: Should we test also `is` and the case where multiple `is` expressions appear in the same branch?

File: plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt [hunk 2]
```diff
@@ -932,6 +951,53 @@ private fun SolverState.checkSimpleConditional(
...
+      if (check.expression is NullExpression) {
+        cont {
+          val complete = solver.booleans {
+            equivalence(solver.boolValue(conditionVar), solver.isNull(subjectVar))
+          }
+          addConstraint(NamedConstraint("$subjectVar is null", complete))
+          NoReturn
+        }
+      } else {
+        val patternName = newName(data.context, "pattern", check.expression)
```
Comment (raulraja) [inline plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt:974]: Probably easier to understand if we extract big branches of code as separate private named functions

**Review by raulraja — APPROVED** [review]
Thanks @serras !

**Pull request merged**